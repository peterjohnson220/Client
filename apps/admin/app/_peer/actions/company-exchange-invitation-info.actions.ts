import { Action } from '@ngrx/store';

export const APPROVE_COMPANY_EXCHANGE_INVITATION =
  '[Peer Admin/Company Exchange Invitation Info] Approve Company Exchange Invitation';
export const APPROVE_COMPANY_EXCHANGE_INVITATION_SUCCESS =
  '[Peer Admin/Company Exchange Invitation Info] Approve Company Exchange Invitation Success';
export const APPROVE_COMPANY_EXCHANGE_INVITATION_ERROR =
  '[Peer Admin/Company Exchange Invitation Info] Approve Company Exchange Invitation Error';
export const DENY_COMPANY_EXCHANGE_INVITATION =
  '[Peer Admin/Company Exchange Invitation Info] Deny Company Exchange Invitation';
export const DENY_COMPANY_EXCHANGE_INVITATION_SUCCESS =
  '[Peer Admin/Company Exchange Invitation Info] Deny Company Exchange Invitation Success';
export const DENY_COMPANY_EXCHANGE_INVITATION_ERROR =
  '[Peer Admin/Company Exchange Invitation Info] Deny Company Exchange Invitation Error';
export const OPEN_COMPANY_INVITATION_INFO =
  '[Peer Admin/Company Exchange Invitation Info] Open Company Invitation Info';
export const CLOSE_COMPANY_INVITATION_INFO =
  '[Peer Admin/Company Exchange Invitation Info] Close Company Invitation Info';

export class ApproveCompanyExchangeInvitation implements Action {
  readonly type = APPROVE_COMPANY_EXCHANGE_INVITATION;

  constructor(public payload: any) {}
}

export class ApproveCompanyExchangeInvitationSuccess implements Action {
  readonly type = APPROVE_COMPANY_EXCHANGE_INVITATION_SUCCESS;
}

export class ApproveCompanyExchangeInvitationError implements Action {
  readonly type = APPROVE_COMPANY_EXCHANGE_INVITATION_ERROR;
}

export class DenyCompanyExchangeInvitation implements Action {
  readonly type = DENY_COMPANY_EXCHANGE_INVITATION;

  constructor(public payload: any) {}
}

export class DenyCompanyExchangeInvitationSuccess implements Action {
  readonly type = DENY_COMPANY_EXCHANGE_INVITATION_SUCCESS;
}

export class DenyCompanyExchangeInvitationError implements Action {
  readonly type = DENY_COMPANY_EXCHANGE_INVITATION_ERROR;
}

export class OpenCompanyInvitationInfo implements Action {
  readonly type = OPEN_COMPANY_INVITATION_INFO;

  constructor(public payload: any) {}
}

export class CloseCompanyInvitationInfo implements Action {
  readonly type = CLOSE_COMPANY_INVITATION_INFO;
}

export type Actions
  = ApproveCompanyExchangeInvitation
  | ApproveCompanyExchangeInvitationSuccess
  | ApproveCompanyExchangeInvitationError
  | DenyCompanyExchangeInvitation
  | DenyCompanyExchangeInvitationSuccess
  | DenyCompanyExchangeInvitationError
  | OpenCompanyInvitationInfo
  | CloseCompanyInvitationInfo;
