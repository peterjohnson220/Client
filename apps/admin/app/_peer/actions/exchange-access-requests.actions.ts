import { Action } from '@ngrx/store';
import { GridDataResult } from '@progress/kendo-angular-grid';

export const LOAD_EXCHANGE_ACCESS_REQUESTS =
  '[Peer Admin/Exchange Access Requests] Load Exchange Access Requests';
export const LOAD_EXCHANGE_ACCESS_REQUESTS_SUCCESS =
  '[Peer Admin/Exchange Access Requests] Load Exchange Access Requests Success';
export const LOAD_EXCHANGE_ACCESS_REQUESTS_ERROR =
  '[Peer Admin/Exchange Access Requests] Load Exchange Access Requests Error';

export class LoadExchangeAccessRequests implements Action {
  readonly type = LOAD_EXCHANGE_ACCESS_REQUESTS;

  constructor(public payload: any) {}
}

export class LoadExchangeAccessRequestsSuccess implements Action {
  readonly type = LOAD_EXCHANGE_ACCESS_REQUESTS_SUCCESS;

  constructor(public payload: GridDataResult) {}
}

export class LoadExchangeAccessRequestsError implements Action {
  readonly type = LOAD_EXCHANGE_ACCESS_REQUESTS_ERROR;
}

export type Actions
  = LoadExchangeAccessRequests
  | LoadExchangeAccessRequestsSuccess
  | LoadExchangeAccessRequestsError;
