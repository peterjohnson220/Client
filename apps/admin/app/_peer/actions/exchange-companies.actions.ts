import { Action } from '@ngrx/store';
import { GridDataResult } from '@progress/kendo-angular-grid';

import { AddExchangeCompaniesRequest } from 'libs/models/peer';

export const LOADING_EXCHANGE_COMPANIES  = '[Peer Admin/Exchange Companies] Loading Exchange Companies';
export const LOADING_EXCHANGE_COMPANIES_SUCCESS  = '[Peer Admin/Exchange Companies] Loading Exchange Companies Success';
export const LOADING_EXCHANGE_COMPANIES_ERROR  = '[Peer Admin/Exchange Companies] Loading Exchange Companies Error';
export const OPEN_ADD_EXCHANGE_COMPANIES_MODAL = '[Peer Admin/Exchange Companies] Open Add Exchange Companies Modal';
export const CLOSE_ADD_EXCHANGE_COMPANIES_MODAL = '[Peer Admin/Exchange Companies] Close Add Exchange Companies Modal';
export const ADDING_EXCHANGE_COMPANIES = '[Peer Admin/Exchange Companies] Adding Exchange Companies';
export const ADDING_EXCHANGE_COMPANIES_SUCCESS = '[Peer Admin/Exchange Companies] Adding Exchange Companies Success';
export const ADDING_EXCHANGE_COMPANIES_ERROR = '[Peer Admin/Exchange Companies] Adding Exchange Companies Error';
export const OPEN_DELETE_EXCHANGE_COMPANY_MODAL = '[Peer Admin/Exchange Companies] Open Delete Exchange Company Modal';
export const CLOSE_DELETE_EXCHANGE_COMPANY_MODAL = '[Peer Admin/Exchange Companies] Close Delete Exchange Company Modal';
export const DELETING_EXCHANGE_COMPANY = '[Peer Admin/Exchange Companies] Deleting Exchange Company';
export const DELETING_EXCHANGE_COMPANY_SUCCESS = '[Peer Admin/Exchange Companies] Deleting Exchange Company Success';
export const DELETING_EXCHANGE_COMPANY_ERROR = '[Peer Admin/Exchange Companies] Deleting Exchange Company Error';

export class LoadingExchangeCompanies implements Action {
  readonly type = LOADING_EXCHANGE_COMPANIES;

  constructor(public payload: any) {}
}

export class LoadingExchangeCompaniesSuccess implements Action {
  readonly type = LOADING_EXCHANGE_COMPANIES_SUCCESS;

  constructor(public payload: GridDataResult) {}
}

export class LoadingExchangeCompaniesError implements Action {
  readonly type = LOADING_EXCHANGE_COMPANIES_ERROR;
}

// Add exchange companies
export class OpenAddExchangeCompaniesModal implements Action {
  readonly type = OPEN_ADD_EXCHANGE_COMPANIES_MODAL;
}

export class CloseAddExchangeCompaniesModal implements Action {
  readonly type = CLOSE_ADD_EXCHANGE_COMPANIES_MODAL;
}

export class AddingExchangeCompanies implements Action {
  readonly type = ADDING_EXCHANGE_COMPANIES;

  constructor(public payload: AddExchangeCompaniesRequest) {}
}

export class AddingExchangeCompaniesSuccess implements Action {
  readonly type = ADDING_EXCHANGE_COMPANIES_SUCCESS;
}

export class AddingExchangeCompaniesError implements Action {
  readonly type = ADDING_EXCHANGE_COMPANIES_ERROR;
}

// Delete Exchange Company
export class OpenDeleteExchangeCompanyModal implements Action {
  readonly type = OPEN_DELETE_EXCHANGE_COMPANY_MODAL;
}

export class CloseDeleteExchangeCompanyModal implements Action {
  readonly type = CLOSE_DELETE_EXCHANGE_COMPANY_MODAL;
}

export class DeletingExchangeCompany implements Action {
  readonly type = DELETING_EXCHANGE_COMPANY;

  constructor(public payload: any) {}
}

export class DeletingExchangeCompanySuccess implements Action {
  readonly type = DELETING_EXCHANGE_COMPANY_SUCCESS;
}

export class DeletingExchangeCompanyError implements Action {
  readonly type = DELETING_EXCHANGE_COMPANY_ERROR;
}

export type Actions
  = LoadingExchangeCompanies
  | LoadingExchangeCompaniesSuccess
  | LoadingExchangeCompaniesError
  | OpenAddExchangeCompaniesModal
  | CloseAddExchangeCompaniesModal
  | AddingExchangeCompanies
  | AddingExchangeCompaniesSuccess
  | AddingExchangeCompaniesError
  | OpenDeleteExchangeCompanyModal
  | CloseDeleteExchangeCompanyModal
  | DeletingExchangeCompany
  | DeletingExchangeCompanySuccess
  | DeletingExchangeCompanyError;
