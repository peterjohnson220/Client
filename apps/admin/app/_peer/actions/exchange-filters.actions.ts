import { Action } from '@ngrx/store';

import { GridDataResult } from '@progress/kendo-angular-grid';

import { ExchangeSearchFilterAggregate } from 'libs/models/peer';

export const LOAD_EXCHANGE_FILTERS = '[Peer Admin/Exchange Filters] Load Exchange Filters';
export const LOAD_EXCHANGE_FILTERS_SUCCESS = '[Peer Admin/Exchange Filters] Load Exchange Filters Success';
export const LOAD_EXCHANGE_FILTERS_ERROR = '[Peer Admin/Exchange Filters] Load Exchange Filters Error';
export const PUT_FILTER = '[Peer Admin/Exchange Filters] Put Filter';
export const PUT_FILTER_SUCCESS = '[Peer Admin/Exchange Filters] Put Filter Success';
export const PUT_FILTER_ERROR = '[Peer Admin/Exchange Filters] Put Filter Error';
export const REORDER_FILTERS = '[Peer Admin/Exchange Filters] Reorder Filters';
export const REORDER_FILTERS_SUCCESS = '[Peer Admin/Exchange Filters] Reorder Filters Success';
export const REORDER_FILTERS_ERROR = '[Peer Admin/Exchange Filters] Reorder Filters Error';

export class LoadExchangeFilters implements Action {
  readonly type = LOAD_EXCHANGE_FILTERS;

  constructor(public payload: any) {}
}

export class LoadExchangeFiltersSuccess implements Action {
  readonly type = LOAD_EXCHANGE_FILTERS_SUCCESS;

  constructor(public payload: GridDataResult) {}
}

export class LoadExchangeFilterError implements Action {
  readonly type = LOAD_EXCHANGE_FILTERS_ERROR;
}

export class PutFilter implements Action {
  readonly type = PUT_FILTER;

  constructor(public payload: ExchangeSearchFilterAggregate) {}
}

export class PutFilterSuccess implements Action {
  readonly type = PUT_FILTER_SUCCESS;

  constructor(public payload: ExchangeSearchFilterAggregate) {}
}

export class PutFilterError implements Action {
  readonly type = PUT_FILTER_ERROR;
}

export class ReorderFilters implements Action {
  readonly type = REORDER_FILTERS;

  constructor(public payload: ExchangeSearchFilterAggregate[]) {}
}

export class ReorderFiltersSuccess implements Action {
  readonly type = REORDER_FILTERS_SUCCESS;
}

export class ReorderFiltersError implements Action {
  readonly type = REORDER_FILTERS_ERROR;
}

export type Actions
  = LoadExchangeFilters
  | LoadExchangeFiltersSuccess
  | LoadExchangeFilterError
  | PutFilter
  | PutFilterSuccess
  | PutFilterError
  | ReorderFilters
  | ReorderFiltersSuccess
  | ReorderFiltersError;
