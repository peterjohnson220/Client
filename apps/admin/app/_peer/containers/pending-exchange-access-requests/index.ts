export * from './pending-exchange-access-requests.component';
