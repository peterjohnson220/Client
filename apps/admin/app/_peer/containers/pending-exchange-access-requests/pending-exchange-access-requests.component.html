<div class="d-flex flex-column h-100">
  <div class="d-flex justify-content-between align-items-center">
    <pf-manage-exchange-section-header
      [sectionTitle]="'Exchange Access Requests'"
      [sectionDescription]="'Listed in the grid below are companies that have requested access to this exchange.'"
    >
    </pf-manage-exchange-section-header>
  </div>
  <div class="d-flex h-100">
    <pf-async-container
      [loading]="pendingExchangeAccessRequestsLoading$ | async"
      [loadingError]="pendingExchangeAccessRequestsLoadingError$ | async"
      [loadingErrorMessage]="'Error loading pending exchange access requests'"
      (reload)="handlePendingExchangeAccessRequestsGridReload()"
      class="flex-grow-1"
    >

      <kendo-grid
        [data]="pendingExchangeAccessRequestsGrid$ | async"
        [scrollable]="true"
      >

        <kendo-grid-column field="CompanyName" title="Company Name"></kendo-grid-column>

        <kendo-grid-column field="CompanyIndustry" title="Industry"></kendo-grid-column>

        <kendo-grid-column field="CompanyLocation" title="Location"></kendo-grid-column>

        <kendo-grid-column field="RequestUser" title="Request User"></kendo-grid-column>

        <kendo-grid-column field="RequestDate" title="Request Date"></kendo-grid-column>

      </kendo-grid>

    </pf-async-container>

  </div>
</div>
