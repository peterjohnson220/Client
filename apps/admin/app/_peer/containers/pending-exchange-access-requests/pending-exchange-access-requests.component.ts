import { Component } from '@angular/core';
import { ActivatedRoute } from '@angular/router';

import { Store } from '@ngrx/store';
import { Observable } from 'rxjs/Observable';
import { GridDataResult } from '@progress/kendo-angular-grid';

import * as fromPeerAdminReducer from '../../reducers';
import * as fromPendingExchangeAccessRequestsActions from '../../actions/pending-exchange-access-requests.actions';

@Component({
  selector: 'pf-pending-exchange-access-requests',
  templateUrl: './pending-exchange-access-requests.component.html',
  styleUrls: ['./pending-exchange-access-requests.component.scss']
})

export class PendingExchangeAccessRequestsComponent {
  pendingExchangeAccessRequestsLoading$: Observable<boolean>;
  pendingExchangeAccessRequestsLoadingError$: Observable<boolean>;
  pendingExchangeAccessRequestsGrid$: Observable<GridDataResult>;
  exchangeId: number;

  constructor(
    private store: Store<fromPeerAdminReducer.State>,
    private route: ActivatedRoute
  ) {
    this.pendingExchangeAccessRequestsLoading$ = this.store.select(fromPeerAdminReducer.getPendingExchangeAccessRequestsLoading);
    this.pendingExchangeAccessRequestsLoadingError$ = this.store.select(fromPeerAdminReducer.getPendingExchangeAccessRequestsLoadingError);
    this.pendingExchangeAccessRequestsGrid$ = this.store.select(fromPeerAdminReducer.getPendingExchangeAccessRequestsGrid);

    this.exchangeId = this.route.snapshot.parent.params.id;
  }

  // Events
  handlePendingExchangeAccessRequestsGridReload() {
    this.store.dispatch(new fromPendingExchangeAccessRequestsActions.LoadPendingExchangeAccessRequests(this.exchangeId));
  }
}
