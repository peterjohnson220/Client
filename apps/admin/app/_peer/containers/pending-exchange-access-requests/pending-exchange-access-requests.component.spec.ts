import { NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';
import { ActivatedRoute } from '@angular/router';

import { StoreModule, Store, combineReducers } from '@ngrx/store';

import * as fromRootState from 'libs/state/state';

import { PendingExchangeAccessRequestsComponent } from './pending-exchange-access-requests.component';
import * as fromPeerAdminReducer from '../../reducers';
import * as fromPendingExchangeAccessRequestsActions from '../../actions/pending-exchange-access-requests.actions';

describe('Pending Exchange Access Requests', () => {
  let fixture: ComponentFixture<PendingExchangeAccessRequestsComponent>;
  let instance: PendingExchangeAccessRequestsComponent;
  let store: Store<fromRootState.State>;
  let activatedRoute: ActivatedRoute;
  let routeIdParam: number;

  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          peerAdmin: combineReducers(fromPeerAdminReducer.reducers)
        })
      ],
      declarations: [
        PendingExchangeAccessRequestsComponent
      ],
      providers: [
        {
          provide: ActivatedRoute,
          useValue: { snapshot: { parent: { params: { id : 1 } } } }
        }
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    store = TestBed.get(Store);
    activatedRoute = TestBed.get(ActivatedRoute);
    routeIdParam = activatedRoute.snapshot.parent.params.id;

    spyOn(store, 'dispatch');

    fixture = TestBed.createComponent(PendingExchangeAccessRequestsComponent);
    instance = fixture.componentInstance;
  });

  it('should match snapshot', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should dispatch a LoadingPendingExchangeAccessRequests action when handlePendingExchangeAccessRequestsGridReload is called', () => {
    const action = new fromPendingExchangeAccessRequestsActions.LoadPendingExchangeAccessRequests(routeIdParam);

    instance.handlePendingExchangeAccessRequestsGridReload();

    expect(store.dispatch).toHaveBeenCalledWith(action);
  });
});
