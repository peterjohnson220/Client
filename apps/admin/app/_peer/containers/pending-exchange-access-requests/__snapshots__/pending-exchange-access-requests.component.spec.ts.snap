// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pending Exchange Access Requests should match snapshot 1`] = `
<pf-pending-exchange-access-requests
  exchangeId={[Function Number]}
  pendingExchangeAccessRequestsGrid$={[Function Store]}
  pendingExchangeAccessRequestsLoading$={[Function Store]}
  pendingExchangeAccessRequestsLoadingError$={[Function Store]}
  route={[Function Object]}
  store={[Function Store]}
>
  <div
    class="d-flex flex-column h-100"
  >
    <div
      class="d-flex justify-content-between align-items-center"
    >
      <pf-manage-exchange-section-header />
    </div>
    <div
      class="d-flex h-100"
    >
      <pf-async-container
        class="flex-grow-1"
      >
        <kendo-grid>
          <kendo-grid-column
            field="CompanyName"
            title="Company Name"
          />
          <kendo-grid-column
            field="CompanyIndustry"
            title="Industry"
          />
          <kendo-grid-column
            field="CompanyLocation"
            title="Location"
          />
          <kendo-grid-column
            field="RequestUser"
            title="Request User"
          />
          <kendo-grid-column
            field="RequestDate"
            title="Request Date"
          />
        </kendo-grid>
      </pf-async-container>
    </div>
  </div>
</pf-pending-exchange-access-requests>
`;
